app/src/main/java/com/architectcoders/arquitectomarvel/ui/detail/CharacterDetailViewModel.kt

```diff
@@ -48,13 +49,13 @@ class CharacterDetailViewModel(
 
     private fun refresh() {
         _model.value = UiModel.Loading
-        loadCharacterId(characterId)
+        _model.value = UiModel.SetCharacterDetails(characterEntity.toDomainCharacter)
```

Comment:
In that case, we won't need SetCharacterDetails class from the sealed class because the activity could get the character data inside the intent